cores/esp8266/WString.h

```diff
@@ -267,8 +249,11 @@ class String {
             if (isSSO()) {
                 setSSO(true); // Avoid emitting of bitwise EXTRACT-AND-OR ops (store-merging optimization)
                 sso.len = len;
-            } else
+                sso.buff[len] = 0;
+            } else {
                 ptr.len = len;
+                ptr.buff[len] = 0;
```

Comment:
Shouldn't there be a check to see if `ptr.buff[len]` is allocated?
If allocation failed, ptr.buff may not be allocated at all, or not being the expected size.
Not sure if it may happen at all in all occurences where `setLen` is called, but maybe best to at least have a check here to see if `buff` isn't a nullptr.